torch/csrc/jit/passes/lower_tuples.cpp

```diff
@@ -40,6 +44,17 @@ static void VisitNode(Node* n, Node* insert_point) {
     }
     return;
   }
+
+  // undo TupleUnpack(TupleIndex)
+  if (n->kind() == prim::TupleIndex) {
+    auto construct = n->inputs().at(0)->node();
+    JIT_ASSERTM(construct->kind() == prim::TupleConstruct, "tuple index not matched to tuple construct");
+    JIT_ASSERT(n->inputs().at(1)->node()->kind() == prim::Constant);
```

Comment:
Should be an attribute not an input for `prim::Constant`, this business of assuming it is a constant will break stuff in the future.